File: 04-match3/match2/src/Tile.py
"""
ISPPV1 2024
Study Case: Match-3

Author: Alejandro Mujica
alejandro.j.mujic4@gmail.com

This file contains the class Tile.
"""

import pygame

import settings


class Tile:
    def __init__(self, x: int, y: int, frame: int) -> None:
        self.x = x
        self.y = y
        self.i = self.y // settings.TILE_SIZE
        self.j = self.x // settings.TILE_SIZE
        self.frame = frame

    def render(self, surface: pygame.Surface) -> None:
        surface.blit(
            settings.TEXTURES["tiles"],
            (self.x + settings.BOARD_OFFSET_X, self.y + settings.BOARD_OFFSET_Y),
            settings.FRAMES["tiles"][self.frame],
        )

File: 7.1-ultimate_fantasy/uf1/src/world/Tile.lua
--[[
    ISPPV1 2023
    Study Case: Ultimate Fantasy (RPG)

    Author: Colton Ogden
    cogden@cs50.harvard.edu

    Modified by Alejandro Mujica (alejandro.j.mujic4@gmail.com)

    This class contains the class Tile.
]]
Tile = Class{}

function Tile:init(x, y, id)
    self.x = x
    self.y = y
    self.id = id
end

function Tile:render(offsetX, offsetY)
    love.graphics.draw(TEXTURES['tiles'], FRAMES['tiles'][self.id],
        (self.x - 1 + offsetX) * TILE_SIZE, (self.y - 1 + offsetY) * TILE_SIZE)
end
